components/ws-manager/pkg/manager/annotations.go

```diff
@@ -171,6 +168,28 @@ type workspaceDisposalStatus struct {
 	BackupComplete bool             `json:"backupComplete,omitempty"`
 	BackupFailure  string           `json:"backupFailure,omitempty"`
 	GitStatus      *csapi.GitStatus `json:"gitStatus,omitempty"`
+	Status         DisposalStatus   `json:"status,omitempty"`
+}
+
+func (m *Manager) markDisposalStatus(ctx context.Context, workspaceID string, disposalStatus *workspaceDisposalStatus) error {
+	b, err := json.Marshal(disposalStatus)
+	if err != nil {
+		return err
+	}
+
+	return m.markWorkspace(ctx, workspaceID, addMark(disposalStatusAnnotation, string(b)))
+}
+
+type DisposalStatus string
+
+const (
+	DisposalStarted  DisposalStatus = "started"
+	DisposalRetrying DisposalStatus = "retrying"
+	DisposalFinished DisposalStatus = "finished"
+)
+
+func (ds DisposalStatus) IsWorking() bool {
```

Comment:
This is a weird function name, that I cannot quite understand.
What does it mean that it is working? Is it currently disposing? I assume that is what this function checks for due to `ds == DisposalFinished` in it, so maybe it would be better to rename this function to be more informative:
`IsDisposing` or `isCurrentlyDisposing`

Reply:
Agree, I was confused by the function name until I read the code logic.

Reply:
👍